lib/miq_expression/field.rb

```diff
@@ -82,8 +82,20 @@ def contains(other)
     )
   end
 
+  def column_type
+    if custom_attribute_column?
+      CustomAttribute.find_by(:name => custom_attribute_column_name).field_type
```

Comment:
In the current implementation we can't really enforce that all the custom attributes with the same name will be with the same type.
I agree adding the resource_type is a good idea and will probably prevent possible failures, adding.